### DISCUSSION TITLE
Impossible GPS coordinates

### DISCUSSION THREAD

**Discussion by Tigwin** [teslamate-org/teslamate#3637] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

GPS sometimes calculates a location that would be impossible, causing map locations to have really long straight lines

### Steps To Reproduce

Check long term data on the map display

### Relevant log output

```shell
See screenshot
```

### Screenshots

<img width="656" alt="image" src="https://github.com/teslamate-org/teslamate/assets/8354498/f3fd1650-6e97-464c-83aa-ac58649ac8e3">

<img width="470" alt="image" src="https://github.com/teslamate-org/teslamate/assets/8354498/41aaf9c5-cc28-4040-962c-9c0ea3a960e1">

### Type of installation

Docker

### Version

1.28.2

**Comment by Tigwin** [comment]
Maybe a cleanup script that checks location records and anything more than 50 miles from the previous can be deleted?

**Comment by cwanja** [comment]
Moving this to a **discussion**, as it is not a TeslaMate issue. TeslaMate records the GPS locations from Tesla API and their API has been known to cause irregular records. 

You can clean up the positions from the `positions` table.
